### PR TITLE
server: improve tsdump

### PR DESCRIPTION
**Pull request by tbg**
- wip
- wipwip
- wip
- x
- wi

Addresses #75993

### CODE REVIEW

**Comment by tbg** [comment]
@rimadeodhar any chance I could hand this off to someone on obs-inf? I won't be able to push this over the finish line for at least a couple of weeks.
I don't think it has to make the release, but investigating tsdumps will work better the moment folks on our end use a binary that includes this PR.

**Comment by rimadeodhar** [comment]
Yes, we can pick this up. Please can you add more details to each individual commit so it would easy for us to pick up the context? Thanks!

File: pkg/kv/kvserver/replica.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ var disableSyncRaftLog = settings.RegisterBoolSetting(
 	"set to true to disable synchronization on Raft log writes to persistent storage. "+
 		"Setting to true risks data loss or data corruption on server crashes. "+
 		"The setting is meant for internal testing only and SHOULD NOT be used in production.",
-	false,
+	true,
```
Comment (tbg) [inline pkg/kv/kvserver/replica.go:89]: This needs to be reverted.

**Comment by tbg** [comment]
Great, thanks! I'm going to leave the commits as is (they weren't structured anyway) but here is a list of the changes from looking at the diff:

----

`tsdump` performs strict checks that verify that it has data for exactly the stores it expects to see according to the node->store mapping. It can happen that there legitimately isn't data there, for example when nodes are down during the tsdump window. We still want the strict checks, but now `tsdump` will defer returning an error and allows restarting with a `-` file to display the data anyway.

The error that results from not having the mapping file in place was improved.

Cluster settings are now set via `SET CLUSTER SETTING` as opposed to overriding settings directly. We're also setting a few more settings that keep the overhead down and speed things up. In particular, we import larger batches and don't sync to disk; we also disable timeseries roll-ups and removal as these may otherwise delete the imported data.

**Comment by rimadeodhar** [comment]
Thanks! Also, is there a tracking issue associated with this? It doesn't seem like there is one but wanted to confirm. I'll create one if there isn't so we can tracking this in our sprint and make sure it is prioritized.

**Comment by tbg** [comment]
No, i created this PR ad hoc when noticing that folks had problems using
tsdump during recent customer incidents.

(sent from mobile, please excuse my brevity)

**Comment by rimadeodhar** [comment]
Closing this out as I've created a new PR with these changes https://github.com/cockroachdb/cockroach/pull/77247.

**Pull request closed without merging**